pkg/vsphere/tasks/waiter.go

```diff
@@ -185,6 +175,43 @@ func IsRetryError(op trace.Operation, err error) bool {
 	}
 }
 
+// Add an additional retry error InvalidArgument
+// This is added for potential race conditions concerning vSAN host partition
+// Normally VM configuration is provided without user input. If there's InvalidArgument fault, it would more possibly be a system error.
+// Details see: https://github.com/vmware/vic/pull/4597
+func IsRetryError(op trace.Operation, err error) bool {
+	return IsTransientError(op, err) || IsInvalidArgumentError(err)
+}
+
+// IsInvalidArgumentError checks if a soap fault or vim fault is InvalidArgument error
+func IsInvalidArgumentError(err error) bool {
+	if soap.IsVimFault(err) {
+		if _, ok := soap.ToVimFault(err).(*types.InvalidArgument); ok {
+			return true
+		}
+		if _, ok := soap.ToVimFault(err).(types.InvalidArgumentFault); ok {
+			return true
+		}
+
+		return false
+	}
+
+	if soap.IsSoapFault(err) {
+		if _, ok := soap.ToSoapFault(err).VimFault().(types.InvalidArgument); ok {
+			return true
+		}
+		if _, ok := soap.ToSoapFault(err).VimFault().(types.InvalidArgumentFault); ok {
+			return true
+		}
+
+		// sometimes we get the correct fault but wrong type
```

Comment:
What does wrong type mean? a pointer type?

Reply:
It's a historic comment made by @caglar10ur a while back. I haven't seen this case yet so I'm not sure exactly. I left the comment there in case it reproduces. 
I'm guessing that sometimes we might get the `InvalidArgument` in error string but the parsed error type is the the expected vim type.